### PR TITLE
Add initial implementation to tail a file or stdin

### PR DESCRIPTION
**Pull request by tuommaki**
As a first part to grab entries from audit.log, implement a `tail`
functionality. The approach chosen here is to follow the file and deliver
string lines via channel. The tail implementation is watching for the file
deletion and in case the file disappears or inode changes it breaks the read
loop and quits program. Intended behavior here is that program gracefully
restarts and starts from beginning when there's a new file to be read.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -2,16 +2,46 @@ package main
...
+		}
+	} else {
+		lines, err = tail.Stdin(logger)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	for s := range lines {
+		fmt.Printf(": %q\n", s)
```
Comment (tuommaki) [inline main.go:41]: Place holder for testing.

File: tail/file.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		return nil, microerror.Mask(err)
+	}
+
+	reader := newLineReader(f)
+
+	err = reader.readLines(output)
+	if err != nil {
+		logger.Errorf(ctx, err, "read")
+		return nil, microerror.Mask(err)
+	}
```
Comment (tuommaki) [inline tail/file.go:35]: If this piece is moved inside the goroutine below, with bit of additional helper code, we can change the behaviour to automatically re-open the new file after log rotation. For now I wanted to keep this simple so I decided to exist on file deletion / change. It has the downside of potentially loosing some metrics during log rotation, but that happens during restarts in our operators as well so I'm not sure if it's _that_ serious. We will find out, I guess and as said above, we can relatively easily change this.

File: tail/file.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+				break
+			}
+
+			err = reader.readLines(output)
+			if err != nil {
+				logger.Errorf(ctx, err, "read")
+				break
+			}
+
+			time.Sleep(100 * time.Millisecond)
```
Comment (tuommaki) [inline tail/file.go:57]: `100ms` here is pretty arbitrary, but it prevents busy loop. Not sure if it should be something like `10ms`.

**Review by whites11 — APPROVED** [review]
seems reasonable.
I'm slightly concerned about the frequent `stat` calls and IOPS.
We need to be sure to monitor disk performance accordingly to avoid problems.

**Comment by tuommaki** [comment]
> I'm slightly concerned about the frequent stat calls and IOPS.

I doubt it would frequently access the actual disk. It returns the file system metadata which should be cached most of the time anyway. [The man-page also suggests this](https://man7.org/linux/man-pages/man2/stat.2.html):

>       Note: for performance and simplicity reasons, different fields in the
>       stat structure may contain state information from different moments
>       during the execution of the system call.  For example, if st_mode or
>       st_uid is changed by another process by calling chmod(2) or chown(2),
>       stat() might return the old st_mode together with the new st_uid, or
>       the old st_uid together with the new st_mode.

**Pull request merged**